include/EntryPointConfirmOptIn.php

```diff
@@ -43,40 +43,104 @@
 /**
  * Class confirm_opt_in
  */
-class ConfirmOptIn
+class EntryPointConfirmOptIn
 {
     /**
      * @var EmailAddress $emailAddress
      */
     private $emailAddress;
+    
+    /**
+     * 
+     * @param array $request
+     * @param array $post
+     */
+    public function __construct($request = null, $post = null) {
+        
+        if(is_null($request)) {
+            $request = $_REQUEST;
+        }
+        
+        if(is_null($post)) {
+            $post = $_POST;
+        }
+        
+        $method = isset($request['method']) && $request['method'] ? $request['method'] : null;
+        
+        switch($method) {
+            case 'confirmOptInSelected':
+                $output = $this->methodConfirmOptInSelected($post);
+                break;
+            default:
+                $output = $this->methodConfirmOptInUser($request);
+                break;
+        }
+        
+        echo $output;
+        sugar_cleanup();
```

Comment:
sugar_cleanup in the constructor makes it harder to customise.  It also unusual to control the process using a constructor. Typically a constructor sets up the state of a class. clean up code would normally be in a destructor.

However, I would be better to call this logic form the entrypoint file. This needs to be in a different class.  So a third party can customise it.

Reply:
I added sugar_cleanup to the entrypoint, instead of the class. 

Calling sugar_cleanup ends the process. The issue here is what happens if I as a third party wish to add more functionality to this code. As soon as I construct the class, it will end the process, preventing me from adding the functionality. 

It it true that it will call the method in this case. However, in order to customise this, I would have to throw away or replace much of this code. Which is less than ideal.

Reply:
If you want to extend the class, use php `extends`. If you want to a custom, use SuiteCRM custom. see previous comment.